tests/unit/server/bundle_manager/__init__.py

```diff
@@ -0,0 +1,216 @@
+            "Test",
+            "User",
+            "password",
+            "Stanford",
+            user_id=self.user_id,
+        )
+
+    def create_make_bundle(self, state=State.MAKING):
+        bundle = MakeBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA_MAKE_BUNDLE,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=state,
+        )
+        return bundle
+
+    def save_bundle(self, bundle):
+        self.bundle_manager._model.save_bundle(bundle)
+
+    def read_bundle(self, bundle, extra_path=""):
+        return open(
+            os.path.join(
+                self.codalab_manager.bundle_store().get_bundle_location(bundle.uuid), extra_path
+            ),
+            "r",
+        )
+
+    def create_run_bundle(self, state=State.CREATED, metadata=None):
+        bundle = RunBundle.construct(
+            targets=[],
+            command='',
+            metadata=dict(BASE_METADATA, **(metadata or {})),
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=state,
+        )
+        return bundle
+
+    def create_bundle_single_dep(
+        self, parent_state=State.READY, bundle_state=State.CREATED, bundle_type=RunBundle
+    ):
+        parent = self.create_run_bundle(parent_state)
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent.uuid), "w+") as f:
+            f.write("hello world")
+        bundle = (
+            self.create_run_bundle(bundle_state)
+            if bundle_type == RunBundle
+            else self.create_make_bundle(bundle_state)
+        )
+        bundle.dependencies = [
+            Dependency(
+                {
+                    "parent_uuid": parent.uuid,
+                    "parent_path": "",
+                    "child_uuid": bundle.uuid,
+                    "child_path": "src",
+                }
+            )
+        ]
+        return bundle, parent
+
+    def create_bundle_two_deps(self):
+        parent1 = self.create_run_bundle(state=State.READY)
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent1.uuid), "w+") as f:
+            f.write("hello world 1")
+        parent2 = self.create_run_bundle(state=State.READY)
+        with open(self.codalab_manager.bundle_store().get_bundle_location(parent2.uuid), "w+") as f:
+            f.write("hello world 2")
+        bundle = MakeBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA_MAKE_BUNDLE,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.STAGED,
+        )
+        bundle.dependencies = [
+            Dependency(
+                {
+                    "parent_uuid": parent1.uuid,
+                    "parent_path": "",
+                    "child_uuid": bundle.uuid,
+                    "child_path": "src1",
+                }
+            ),
```

Comment:
Could we wrap this in a function so we don't have raw strings?  Ideally this should be a dataclass...

Reply:
`Dependency` is an ORMObject (https://github.com/codalab/codalab-worksheets/blob/a2306fac3d4a6cfbb2d4172f19a6cf9f0c37cbf1/codalab/model/orm_object.py#L44), and I just followed the way its constructor works, so if we wanted to change the way it's constructed, we could probably do that, though I think that's beyond the scope of this PR.